File: CSharp/modul5_kel46.cs
using System;

namespace Modul5
{
    class Program
    {
        // main function
        static void Main(string[] args)
        {
            // deklarasi variabel
            string email, password;

            Console.Write("Email : ");
            // membaca input user dan disimpan dalam variabel email
            email = Console.ReadLine();
            Console.Write("Password : ");
            // membaca input user dan disimpan dalam variabel password
            password = Console.ReadLine();

            // membuat object baru dengan variabel data dan menginisialisasi constructor userService
            // dengan isi parameter input email dan password
            userService data = new userService(email, password);
            // memanggil function login
            data.login();

            Console.ReadKey();
        }
    }

    // class userService
    class userService {
        // deklarasi variabel untuk class userService
        public string[,] data; // varibel data dengan tipe data string array 2 dimensi
        private string email, password, roles = ""; 

        // constructor yang akan diinisialisasi dengan parameter emails dan password bertipe string
        public userService(string emails, string passwords)
        {
            // menset email dan password sesuai dengan isi parameter
            email = emails;
            password = passwords;
            // menset data dengan array 2 dimensi berisi email, password, dan role user yang bisa login
            data = new string[2, 3] {
                {"soffankelompok46@gmail.com", "12345", "superadmin" },
                {"firmankelompok46@gmail.com", "12345", "user"  }
            };
        }

        // function login
        public void login()
        {
            // memanggil function checkCredentials dan disimpan dalam variabel status, role
            var (status,role) = checkCredentials();
            // jika status sama dengan true
            if (status == false)
            {
                // maka akan menampilkan welcome role, dan email user yang login
                Console.WriteLine("\nWelcome " + role);
                Console.WriteLine("Logged it as user email: " + email);
            }
            else
            {
                // jika bernilai false maka akan menampilkan invalid login
                Console.WriteLine("\nInvalid Login");
            }
        }

        // function checkCredentials yang akan mengembalikan boolean dan string dengan modifier private
        private (bool,string) checkCredentials()
        {
            // perulangan for dengan inisialisasi i = 0, range i kurang dari jumlah data, dan increment i++
            // untuk mengecek data yang dibuat dengan input email dan password
            for (int i = 0; i < data.GetLength(0); i++)
            {
                // jika data sama dengan input email dan password sama dengan input password
                if (data[i, 0] == email && data[i, 1] == password)
                {
                    // akan menyimpan role didalam variabel roles
                    roles = data[i, 2];
                    // dan mengembalikan nilai true dan role user yang login
                    return (true,roles);
                }
            }
            // jika gagal maka akan mengembalikan boolean false dan roles
            return (false, roles);
        }
    }
}

File: Php/modul5_kel46.php
<!DOCTYPE html>
<html lang="en">
    <head>
        <title>Class and Constructor</title>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
    </head>
    <body>
    <!-- form html dengan action ke file itu sendiri dan method POST -->
     <form action="modul5_kel46.php" method="POST">
        <label >System Login Kelompok 46</label><br>
        <!-- input email dengan type email dan input password dengan type password dan button submit,  -->
        <input type="email" name="email" placeholder="email">
        <input type="password" name="password" placeholder="password">
        <button type="submit">Submit</button>
     </form>
    </body>
</html>

<?php
// include file userService.php agar bisa menggunakan method yang ada didalam class userService
include("userService.php");

// mengecek apakah hasil post email dan password tidak kosong
if (!empty($_POST['email']) && $_POST['password']) {
    // membuat constructor baru untuk userService dengan variabel user
    $user = new userService($_POST['email'], $_POST['password']);

    // memanggil function login dan mengecek apakah email dan password sudah sesuai dengan data
    if($get_user = $user->login()) {
        // menampilkan pesan welcome role dan email user yang login
        echo 'Welcome '.$user->getRole();
        echo ', Logged it as user email:'.$get_user;
    } else {
        // jika login gagal atau email dan password tidak sesuai akan menampilkan invalid login
        echo 'Invalid Login';
    }
}

?>